babel/numbers.py

```diff
@@ -1112,6 +1112,11 @@ def apply(
             retval = retval.replace(u'¤¤', currency.upper())
             retval = retval.replace(u'¤', get_currency_symbol(currency, locale))
 
+        # remove single quotes around text, except for doubled single quotes
+        # which are replaced with a single quote
+        retval = re.sub(r"(^|[^'])'(?!')|'(?<!')($|[^'])", r'\1\2', retval)
+        retval = retval.replace("''", "'")
```

Comment:
I think the regular expression is too complex and buggy like:

```pycon
>>> numbers.format_decimal(42, "'$'''0", locale='en')
"$''42"  # shoud be "$'42"
```

```suggestion
        retval = re.sub(r"'([^']*)'", lambda m: m.group(1) or "'", retval)
```

Reply:
Looks good :+1: